carp.client.core/src/commonMain/kotlin/dk/cachet/carp/client/domain/StudyRuntime.kt

```diff
@@ -29,6 +29,10 @@ class StudyRuntime private constructor(
          * Instantiate a [StudyRuntime] by registering the client device in the [deploymentService].
          * In case the device is immediately ready for deployment, also deploy.
          *
+         * @throws IllegalArgumentException when:
+         * - a deployment with [studyDeploymentId] does not exist
+         * - [deviceRoleName] is not present in the deployment or is already registered and a different [deviceRegistration] is specified than a previous request
+         * - [deviceRegistration] is invalid for the specified device or uses a device ID which has already been used as part of registration of a different device
          * @throws UnsupportedOperationException in case deployment failed since not all necessary plugins to execute the study are available.
          */
         internal suspend fun initialize(
```

Comment:
How will `unregister` come into play with the StudyRuntime? Do you have to unregister a device before you initialize a StudyRuntime with a new deviceRegistration?

Reply:
Good question. Currently, seems like it, yes. But something to discuss what a good API would be during the meeting.

Reply:
Yes agree, it is not important for this PR. Still wanted to ask the question so we can agree it is something we will look at in the future.